### PR TITLE
More coverage

### PR DESCRIPTION
**Pull request by bramtayl**
I think hope should just about get us to 100% coverage. I expect the remaining uncovered lines will be false negatives.

Aside from expanding tests, I've also:

- Fixed some typos in the error messages
- Moved some of the error handling into separate functions for easier testing.

### CODE REVIEW

**Comment by bramtayl** [comment]
Ok, so,

- In the first round I added more tests to basically cover everything
- This led to many false negatives (lines that were run but not marked as covered)
- Looking through these false negatives, I suspected that perhaps the reason was related to intermediate assignments and terse function definitions.
- I removed these, but it didn't seem to make much difference.
- Then I realized the real problem was macros which introduced new line number information, namely, `@locked` and `@stderr_as_debug`. I just removed the `@locked` macro: the lock do block syntax is almost as terse, and makes the reliance on a global constant more explicit. I had to modify `@stderr_as_debug` manually to fix the line number information.

In short, I think the diff here is perhaps more extensive than necessary, but if there were unnecessary changes, they were stylistic.

**Comment by bramtayl** [comment]
See also https://discourse.julialang.org/t/nagging-coverage-question/61821

**Comment by bramtayl** [comment]
Ok, updates, the macro coverage issue in Base seems to be a bug, see https://github.com/JuliaLang/julia/issues/41043, so I reverted the renumbering changes. Also, it's been a week, so I'll merge if this passes.
